File: backend/src/models/Challenge.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Challenge = sequelize.define(
  'Challenge',
  {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true,
    },
    title: {
      type: DataTypes.STRING,
      allowNull: false,
      validate: {
        notEmpty: true,
      },
    },
    description: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
    startDate: {
      type: DataTypes.DATE,
      allowNull: false,
      field: 'start_date',
    },
    endDate: {
      type: DataTypes.DATE,
      allowNull: false,
      field: 'end_date',
    },
    status: {
      type: DataTypes.ENUM('active', 'inactive', 'completed'),
      defaultValue: 'active',
    },
  },
  {
    tableName: 'challenges',
    timestamps: true,
    underscored: true,
  }
);

module.exports = Challenge;

File: frontend-app/lib/features/challenges/Models/challenge.dart
class Challenge {
  final String id;
  final String title;
  final String description;
  final String status;
  final String createdAt;

  Challenge({
    required this.id,
    required this.title,
    required this.description,
    required this.status,
    required this.createdAt,
  });

  factory Challenge.fromJson(Map<String, dynamic> json) => Challenge(
        id: json['id'],
        title: json['title'],
        description: json['description'],
        status: json['status'],
        createdAt: json['createdAt'],
      );
}
